perception/autoware_compare_map_segmentation/src/distance_based_compare_map_filter/node.cpp

```diff
@@ -32,8 +32,7 @@ void DistanceBasedStaticMapLoader::onMapCallback(
   pcl::PointCloud<pcl::PointXYZ> map_pcl;
   pcl::fromROSMsg<pcl::PointXYZ>(*map, map_pcl);
   const auto map_pcl_ptr = pcl::make_shared<pcl::PointCloud<pcl::PointXYZ>>(map_pcl);
-
-  (*mutex_ptr_).lock();
+  std::unique_lock<std::mutex> lock(static_map_loader_mutex_);
```

Comment:
Why do you use `unique_lock` here?

Reply:
to protect `tree_` as expalained at https://github.com/autowarefoundation/autoware.universe/pull/9097#discussion_r1807811815

Reply:
I meant "why did you changed `lock_guard` to `unique_lock` ? ".

Reply:
it was my mistake, Since it no need to set unlock manually in this case, so I change back to lock_guard for more efficiency. Thanks,